File: Trabalho2/Python/complexos(1).py
class Complexos:
    def __init__(self, real: float, imag: float):
        self.real = real
        self.imag = imag

    def Soma(self, outro):
        parteReal = self.real + outro.real
        parteImaginaria = self.imag + outro.imag
        return Complexos(parteReal, parteImaginaria)
    
    def Subtracao(self, outro):
        parteReal = self.real - outro.real
        parteImaginaria = self.imag - outro.imag
        return Complexos(parteReal, parteImaginaria)
    
    def Divisao(self, outro):
        denominaodor = outro.real * outro.real + outro.imag * outro.imag
        parteReal = (outro.real * self.real + outro.imag * self.imag)/denominaodor
        parteImaginaria = (outro.real * self.imag - self.real * outro.imag)/denominaodor
        return Complexos(parteReal, parteImaginaria)
    
    def Multiplicacao(self, outro):
        parteReal = self.real * outro.real - self.imag * outro.imag
        parteImaginaria = self.real * outro.imag + self.imag * outro.real
        return Complexos(parteReal, parteImaginaria)

    def getReal(self):
        return self.real
    
    def getImag(self):
        return self.imag

File: Trabalho2/Rust/complexos(1).rs
pub struct Complexos {
    real: f64,
    imaginario: f64,
}

impl Complexos {
    pub fn novo(real: f64, imaginario: f64) -> Complexos {
        Complexos {
            real: real,
            imaginario: imaginario,
        }
    }

    pub fn getReal(&self) -> f64 {
        self.real
    }

    pub fn getImaginario(&self) -> f64 {
        self.imaginario
    }

    pub fn soma(&self, complexo: &Complexos) -> Complexos {
        Complexos {
            real: self.real + complexo.getReal(),
            imaginario: self.imaginario + complexo.getImaginario(),
        }
    }

    pub fn subtracao(&self, complexo: &Complexos) -> Complexos {
        Complexos {
            real: self.real - complexo.getReal(),
            imaginario: self.imaginario - complexo.getImaginario(),
        }
    }

    pub fn multiplicacao(&self, complexo: &Complexos) -> Complexos {
        Complexos {
            real: self.real * complexo.getReal() - self.imaginario * complexo.getImaginario(),
            imaginario: self.real * complexo.getImaginario() + self.imaginario * complexo.getReal(),
        }
    }

    pub fn divisao(&self, complexo: &Complexos) -> Complexos {
        Complexos {
            real: (self.real * complexo.getReal() + self.imaginario * complexo.getImaginario())
                / (complexo.getReal().powi(2) + complexo.getImaginario().powi(2)).sqrt(),
            imaginario: (self.imaginario * complexo.getReal() - self.real * complexo.getImaginario())
                / (complexo.getReal().powi(2) + complexo.getImaginario().powi(2)).sqrt(),
        }
    }
}